crates/meilisearch/src/routes/indexes/settings.rs

```diff
@@ -716,7 +716,32 @@ pub async fn delete_all(
 
 fn validate_settings(
     settings: Settings<Unchecked>,
-    _index_scheduler: &IndexScheduler,
+    index_scheduler: &IndexScheduler,
 ) -> Result<Settings<Unchecked>, ResponseError> {
+    use meilisearch_types::milli::update::Setting;
+    use meilisearch_types::milli::vector::settings::EmbedderSource;
+
+    let features = index_scheduler.features();
+    if let Setting::Set(embedders) = &settings.embedders {
+        for SettingEmbeddingSettings { inner: embedder } in embedders.values() {
+            let Setting::Set(embedder) = embedder else {
+                continue;
+            };
+            if let Setting::Set(source) = embedder.source {
+                if source == EmbedderSource::Composite {
+                    features.check_composite_embedders("using `\"composite\"` as source")?;
+                }
+            }
```

Comment:
Small style change if you prefer, I’m not especially attached to it, so do whatever you want with it 😁 

```suggestion
            if matches!(embedder.source, Setting::Set(EmbedderSource::Composite)) {
                features.check_composite_embedders("using `\"composite\"` as source")?;
            }
```

Reply:
Applied, changed the `if let` afterwards as well for consistency.